pkg/common/cli/trust_domain.go

```diff
@@ -1,5 +1,30 @@
 package cli
 
-// MaxTrustDomainLength is the maximum length of a trust domain according
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+)
+
+// maxTrustDomainLength is the maximum length of a trust domain according
 // to the SPIFFE standard.
-const MaxTrustDomainLength = 255
+const maxTrustDomainLength = 255
+
+// ParseTrustDomain parses a configured trustDomain in a consistent way
+// for either the SPIRE agent or server.
+func ParseTrustDomain(trustDomain string, logger logrus.FieldLogger) (td spiffeid.TrustDomain, err error) {
+	// Warn on a non-conforming trust domain to avoid breaking backwards compatibility
+	if len(trustDomain) > maxTrustDomainLength {
```

Comment:
The final nitpick, I promise! We should do this length check on `td.String()` instead of `trustDomain`, since technically, they can provide a trust domain ID in the configuration (e.g. `spiffe://example.org`), which is parsed and handled by `spiffeid.TrustDomainFromString()`. In this instance, `td.String()` will return `"example.org"`. Technically if we keep it as is we might warn on a trust domain name that is actually smaller than 256.

Reply:
Good catch!